### PR TITLE
Migrate project card 

### PR DESCRIPTION
**Pull request by pixelfact**

### CODE REVIEW

File: components/ds/Skeleton/Skeleton.tsx [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+export default function SkeletonEl({ variant = "rectangular", color = "blue", ...props }: SkeletonProps) {
+  switch (variant) {
+    case "rectangular":
+      return <SkeletonRectangular {...props} color={color} />;
+    case "circular":
+      return <SkeletonCircular {...props} color={color} />;
+    case "rounded":
+      return <SkeletonRounded {...props} color={color} />;
+    default:
+      return null;
```
Comment (haydencleary) [inline components/ds/Skeleton/Skeleton.tsx:15]: Shouldn't the default be `<SkeletonRectangular>` ? You could make this a polymorphic component and remove the switch/case.

File: components/ds/Skeleton/components/Circular.tsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
```
Comment (haydencleary) [inline components/ds/Skeleton/components/Circular.tsx:1]: Don't need to import React

File: components/ds/Skeleton/components/Rectangular.tsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
```
Comment (haydencleary) [inline components/ds/Skeleton/components/Rectangular.tsx:1]: Don't need to import React

File: components/ds/Skeleton/components/Rounded.tsx [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import React from "react";
```
Comment (haydencleary) [inline components/ds/Skeleton/components/Rounded.tsx:1]: Don't need to import React

File: components/ds/Sort/Sort.tsx [hunk 5]
```diff
@@ -0,0 +1,68 @@
+import { SortProps } from "./Sort.type.ts";
+import { Listbox, Transition } from "@headlessui/react";
+import Arrow from "../../../src/assets/icons/Arrow.tsx";
+import ArrowDownSLine from "../../../src/icons/ArrowDownSLine.tsx";
+import CheckLine from "../../../src/icons/CheckLine.tsx";
```
Comment (haydencleary) [inline components/ds/Sort/Sort.tsx:5]: Use `<Icon>`

File: components/ds/Sort/index.ts
Comment (haydencleary) [inline components/ds/Sort/index.ts:1]: Same remark for barrel files

File: components/ds/tag/tag.loading.tsx [hunk 6]
```diff
@@ -0,0 +1,22 @@
+import { cn } from "src/utils/cn";
+import React from "react";
```
Comment (haydencleary) [inline components/ds/tag/tag.loading.tsx:2]: Don't need to import React

File: components/ds/tag/tag.stories.tsx [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+import TagLoading from "@/components/ds/tag/tag.loading.tsx";
+
+const defaultProps: TagProps = {
+  children: <span>Tag</span>,
+};
+
+const defaultPropsWithIcon = (orange?: boolean): TagProps => ({
+  children: (
+    <>
+      <User3Line className={orange ? "text-orange-500" : ""} />
```
Comment (haydencleary) [inline components/ds/tag/tag.stories.tsx:15]: Use `<Icon>`

File: components/ds/tag/tag.tsx [hunk 8]
```diff
@@ -1,19 +1,19 @@
 import { cn } from "src/utils/cn";
-import { ElementType, PropsWithChildren } from "react";
-import { tagVariants, TagVariants } from "./tag.variants.ts";
+import React from "react";
```
Comment (haydencleary) [inline components/ds/tag/tag.tsx:2]: Don't need to import React

File: components/ds/thumbnail-group/index.ts
Comment (haydencleary) [inline components/ds/thumbnail-group/index.ts:1]: Same remark for barrel files

File: components/ds/thumbnail-group/thumbnail-group.loading.tsx [hunk 9]
```diff
@@ -0,0 +1,16 @@
+import { ThumbnailGroupLoadingProps } from "./thumbnail-group.type.ts";
+import { cn } from "../../../src/utils/cn.ts";
```
Comment (haydencleary) [inline components/ds/thumbnail-group/thumbnail-group.loading.tsx:2]: Alias

File: components/ds/thumbnail-group/thumbnail-group.loading.tsx [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+import { thumbnailGroupVariant } from "@/components/ds/thumbnail-group/thumbnail-group.variant.ts";
+import React from "react";
```
Comment (haydencleary) [inline components/ds/thumbnail-group/thumbnail-group.loading.tsx:4]: Don't need to import React

File: components/ds/thumbnail-group/thumbnail-group.tsx [hunk 11]
```diff
@@ -0,0 +1,16 @@
+import { cn } from "../../../src/utils/cn.ts";
+import { ThumbnailGroupProps } from "@/components/ds/thumbnail-group/thumbnail-group.type.ts";
+import React from "react";
+import { Thumbnail } from "@/components/ds/thumbnail";
+import { thumbnailGroupVariant } from "@/components/ds/thumbnail-group/thumbnail-group.variant.ts";
+export function ThumbnailGroup({ thumbnails, className, defaultSrc = true, ...props }: ThumbnailGroupProps) {
+  return (
+    <div className={cn(thumbnailGroupVariant({ ...props }), className)}>
+      {thumbnails.map(thumbnail => (
+        <Thumbnail {...thumbnail} {...props} defaultSrc={defaultSrc} />
```
Comment (haydencleary) [inline components/ds/thumbnail-group/thumbnail-group.tsx:10]: Missing key prop

File: components/ds/thumbnail-group/thumbnail-group.tsx [hunk 12]
```diff
@@ -0,0 +1,16 @@
+import { cn } from "../../../src/utils/cn.ts";
```
Comment (haydencleary) [inline components/ds/thumbnail-group/thumbnail-group.tsx:1]: Alias

File: components/ds/thumbnail-group/thumbnail-group.tsx [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+import { ThumbnailGroupProps } from "@/components/ds/thumbnail-group/thumbnail-group.type.ts";
+import React from "react";
```
Comment (haydencleary) [inline components/ds/thumbnail-group/thumbnail-group.tsx:3]: Don't need to import React

File: components/ds/thumbnail/index.ts
Comment (haydencleary) [inline components/ds/thumbnail/index.ts:1]: Same remark for barrel files

File: components/ds/thumbnail/thumbnail.loading.tsx [hunk 14]
```diff
@@ -0,0 +1,30 @@
+import { ThumbnailLoadingProps } from "./thumbnail.type.ts";
+import { cn } from "../../../src/utils/cn.ts";
```
Comment (haydencleary) [inline components/ds/thumbnail/thumbnail.loading.tsx:2]: Alias

File: components/ds/thumbnail/thumbnail.tsx [hunk 15]
```diff
@@ -0,0 +1,18 @@
...
+import { IMAGES } from "../../../src/assets/img";
+import GalleryLine from "../../../src/assets/icons/GalleryLine.tsx";
+import { thumbnailVariant } from "@/components/ds/thumbnail/thumbnail.variant.ts";
+export function Thumbnail({ src, alt, className, defaultSrc = true, ...props }: ThumbnailProps) {
+  return (
+    <div className={cn(thumbnailVariant({ ...props }), className)}>
+      {src || defaultSrc ? (
+        <img src={src || IMAGES.logo.space} alt={alt} className="h-full w-full object-cover" loading="lazy" />
+      ) : (
+        <GalleryLine className="h-4 w-4 text-spaceBlue-300" />
```
Comment (haydencleary) [inline components/ds/thumbnail/thumbnail.tsx:12]: Use `<Icon>`

File: components/ds/thumbnail/thumbnail.tsx [hunk 16]
```diff
@@ -0,0 +1,18 @@
+import { ThumbnailProps } from "./thumbnail.type.ts";
+import { cn } from "../../../src/utils/cn.ts";
+import { IMAGES } from "../../../src/assets/img";
+import GalleryLine from "../../../src/assets/icons/GalleryLine.tsx";
```
Comment (haydencleary) [inline components/ds/thumbnail/thumbnail.tsx:4]: Alias

File: components/features/contributor/contributor.tsx [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+      </span>
+
+      {isRegistered ? (
+        <img
+          id={`od-logo-${login}`}
+          src={IMAGES.logo.original}
+          className="w-3.5"
+          loading="lazy"
+          alt="OnlyDust"
+          {...withTooltip(T("contributor.table.userRegisteredTooltip"), { className: "w-36" })}
```
Comment (haydencleary) [inline components/features/contributor/contributor.tsx:52]: Try using `<Translate>`

File: components/features/contributors-avatars/contributors-avatars.tsx [hunk 18]
```diff
@@ -0,0 +1,30 @@
+import { ContributorsAvatarsProps } from "./contributors-avatars.type.ts";
+import { TooltipPosition, withTooltip } from "../../../src/components/Tooltip";
```
Comment (haydencleary) [inline components/features/contributors-avatars/contributors-avatars.tsx:2]: Alias

File: components/features/contributors-avatars/contributors-avatars.tsx [hunk 19]
```diff
@@ -0,0 +1,30 @@
...
+import { ThumbnailGroup } from "@/components/ds/thumbnail-group";
+import React from "react";
```
Comment (haydencleary) [inline components/features/contributors-avatars/contributors-avatars.tsx:4]: Don't need to import React

File: components/features/contributors-avatars/index.ts
Comment (haydencleary) [inline components/features/contributors-avatars/index.ts:1]: Same barrel file remark

File: components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx [hunk 20]
```diff
@@ -0,0 +1,66 @@
+import { useMediaQuery } from "usehooks-ts";
+import { viewportConfig } from "../../../src/config.ts";
+import { cn } from "../../../src/utils/cn.ts";
+import { Button } from "../../ds/button/button.tsx";
+import CheckLine from "../../../src/icons/CheckLine.tsx";
```
Comment (haydencleary) [inline components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx:5]: Alias

File: components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx [hunk 21]
```diff
@@ -0,0 +1,66 @@
...
+  //
+  // useMutationAlert({
+  //   mutation: rest,
+  //   success: {
+  //     message: T("projectLeadInvitation.success", { projectName }),
+  //   },
+  //   error: {
+  //     default: true,
+  //   },
+  // });
```
Comment (haydencleary) [inline components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx:37]: Can this be removed ?

File: components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx [hunk 22]
```diff
@@ -0,0 +1,66 @@
...
+import Translate from "@/components/layout/translate/translate.tsx";
+import { ProjectLeadInvitationProps } from "@/components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.type.ts";
+
+export default function ProjectLeadInvitationBanner({
+  isLoading,
+  onClick,
+  projectName,
+  btnLabel,
+  size,
+  on = "cards",
```
Comment (haydencleary) [inline components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx:17]: Not a clear prop name, what is this for ?

File: components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx [hunk 23]
```diff
@@ -0,0 +1,66 @@
...
+      return <Translate token={"projectLeadInvitation.accept"} />;
+    }
+    return <Translate token={"projectLeadInvitation.acceptShort"} />;
+  }, [btnLabel]);
+
+  return (
+    <div className={cn(base())}>
+      <div className={cn(title())}>{TitleContent}</div>
+      <Button size={size} onClick={onClick} disabled={isLoading}>
+        {size === "l" ? <CheckLine className="text-xl font-normal text-black" /> : null}
```
Comment (haydencleary) [inline components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx:61]: Use `<Icon>`

File: components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.variant.ts [hunk 24]
```diff
@@ -0,0 +1,35 @@
+import { tv, VariantProps } from "tailwind-variants";
+import { cn } from "../../../src/utils/cn.ts";
```
Comment (haydencleary) [inline components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.variant.ts:2]: Alias

File: components/features/project-missing-github-banner/project-missing-github-banner.tsx [hunk 25]
```diff
@@ -0,0 +1,19 @@
...
+import { Button } from "@/components/ds/button/button.tsx";
+export function ProjectMissingGithubBanner({ slug = "" }: { slug: string }) {
+  const { T } = useIntl();
+
+  return (
+    <div className="bg-oritems-center flex items-center justify-between gap-3 rounded-2xl bg-orange-800 p-3">
+      <p className="font-walsheim text-sm font-medium text-white">{T("project.missingGithubAppInstall.message")}</p>
+      <Link href={generatePath(RoutePaths.ProjectDetailsEditRepos, { projectKey: slug })}>
+        <Button size="s" className="whitespace-nowrap" type="secondary" accentColor="orange">
+          {T("project.missingGithubAppInstall.button")}
```
Comment (haydencleary) [inline components/features/project-missing-github-banner/project-missing-github-banner.tsx:14]: Use `<Translate>`

File: components/features/project-missing-github-banner/project-missing-github-banner.tsx [hunk 26]
```diff
@@ -0,0 +1,19 @@
...
+import { RoutePaths } from "src/App";
+import { useIntl } from "src/hooks/useIntl";
+import Link from "next/link";
+import { Button } from "@/components/ds/button/button.tsx";
+export function ProjectMissingGithubBanner({ slug = "" }: { slug: string }) {
+  const { T } = useIntl();
+
+  return (
+    <div className="bg-oritems-center flex items-center justify-between gap-3 rounded-2xl bg-orange-800 p-3">
+      <p className="font-walsheim text-sm font-medium text-white">{T("project.missingGithubAppInstall.message")}</p>
```
Comment (haydencleary) [inline components/features/project-missing-github-banner/project-missing-github-banner.tsx:11]: Use `<Translate>`

File: components/layout/pages/scroll-view/index.ts
Comment (haydencleary) [inline components/layout/pages/scroll-view/index.ts:1]: Same barrel file remark

File: components/layout/pages/scroll-view/scroll-view.tsx [hunk 27]
```diff
@@ -0,0 +1,22 @@
+import { ScrollViewProps } from "./scroll-view.type.ts";
+import { ForwardedRef, forwardRef } from "react";
+import { cn } from "../../../../src/utils/cn.ts";
```
Comment (haydencleary) [inline components/layout/pages/scroll-view/scroll-view.tsx:3]: Alias

File: components/layout/pages/space-background/index.ts
Comment (haydencleary) [inline components/layout/pages/space-background/index.ts:1]: Same barrel file remark

File: components/layout/pages/with-background/index.ts
Comment (haydencleary) [inline components/layout/pages/with-background/index.ts:1]: Same barrel file remark

File: components/layout/pages/with-background/with-background.tsx [hunk 28]
```diff
@@ -0,0 +1,18 @@
+import { WithBackgroundProps } from "./with-background.type.ts";
+import { cn } from "tailwind-variants";
+
+export function PageWithBackground(props: WithBackgroundProps) {
```
Comment (haydencleary) [inline components/layout/pages/with-background/with-background.tsx:4]: I think you forgot to destructure the props here ?

File: components/layout/rounded-image/rounded-image.tsx [hunk 29]
```diff
@@ -0,0 +1,63 @@
+import { tv, VariantProps } from "tailwind-variants";
+import { cn } from "src/utils/cn";
+import { IMAGES } from "../../../src/assets/img";
```
Comment (haydencleary) [inline components/layout/rounded-image/rounded-image.tsx:3]: Alias

File: components/layout/translate/translate.tsx [hunk 30]
```diff
@@ -2,12 +2,12 @@
...
+export type TranslateProps = {
   token: Key;
   params?: { [key: string]: string | number };
 };
 
-export default function Translate({ token, params }: ClientIntlProps) {
+export default function Translate({ token, params }: TranslateProps) {
   const { T } = useIntl();
 
   return <>{T(token, params)}</>;
```
Comment (haydencleary) [inline components/layout/translate/translate.tsx:13]: Is the fragment necessary ?

File: tools/create-components.ts [hunk 31]
```diff
@@ -0,0 +1,155 @@
...
+
+function kebabToPascal(str: string) {
+  const words = str.split("-");
+  const pascalCaseWords = words.map(word => word.charAt(0).toUpperCase() + word.slice(1));
+  return pascalCaseWords.join("");
+}
+
+function kebabToCamel(str: string) {
+  const words = str.split("-");
+  const pascalCaseWords = words.map(word => word.charAt(0).toLowerCase() + word.slice(1));
```
Comment (haydencleary) [inline tools/create-components.ts:29]:
```suggestion
  const camelCaseWords = words.map(word => word.charAt(0).toLowerCase() + word.slice(1));
```

File: tools/create-components.ts [hunk 32]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+async function createComponents() {
+  const { folder, name, path, loading, variant } = await promptName();
+  await fs.mkdir(path);
+  await createFiles({
+    folder,
+    name,
+    path,
+    PascalName: kebabToPascal(name),
```
Comment (haydencleary) [inline tools/create-components.ts:149]: I'd call this `pascalName` even though it doesn't respect the style of pascal case

**Review by haydencleary — COMMENTED** [review]
In the `/components/ds` folder some components have capital letters and some don't.

I know I repeated lots of "Alias", "React import" and others. It's just so they don't get forgotten ^^

File: components/features/project-missing-github-banner/project-missing-github-banner.tsx [hunk 33]
```diff
@@ -0,0 +1,21 @@
...
+import { TProjectMissingGithubBanner } from "./project-missing-github-banner.types";
+
+export function ProjectMissingGithubBanner({ slug = "" }: TProjectMissingGithubBanner.Props) {
+  const { T } = useIntl();
+
+  return (
+    <div className="bg-oritems-center flex items-center justify-between gap-3 rounded-2xl bg-orange-800 p-3">
+      <p className="font-walsheim text-sm font-medium text-white">{T("project.missingGithubAppInstall.message")}</p>
+      <Link href={generatePath(RoutePaths.ProjectDetailsEditRepos, { projectKey: slug })}>
+        <Button size="s" className="whitespace-nowrap" type="secondary" accentColor="orange">
```
Comment (NeoxAzrot) [inline components/features/project-missing-github-banner/project-missing-github-banner.tsx:15]: Need onclick

File: app/migration/projects/context/project.context.tsx
[the remark this answers is not in the document]
Comment (NeoxAzrot) [reply app/migration/projects/context/project.context.tsx:1]: @alexbeno

File: app/migration/projects/context/project.context.tsx [hunk 34]
```diff
@@ -0,0 +1,111 @@
...
+  }
+  function onClearFilter() {
+    setFilters(DEFAULT_PROJECTS_FILTER);
+    setStorage(DEFAULT_PROJECTS_FILTER);
+  }
+
+  /** need this to migrate existing filter for sponsor */
+  useEffect(() => {
+    if (storage && storage.sponsors && storage.sponsors.length > 0) {
+      const anySponsor = storage.sponsors as unknown as any;
```
[the remark this answers is not in the document]
Comment (NeoxAzrot) [reply app/migration/projects/context/project.context.tsx:73]: @alexbeno

File: app/migration/projects/context/project.context.type.ts [hunk 35]
```diff
@@ -0,0 +1,57 @@
...
+
+export interface ProjectFilter {
+  ownership: Ownership;
+  technologies: string[];
+  sponsors: string[];
+  search?: string;
+  sorting: Sorting;
+}
+
+export enum Sorting {
```
[the remark this answers is not in the document]
Comment (NeoxAzrot) [reply app/migration/projects/context/project.context.type.ts]: @alexbeno

File: components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx (hunk 22 above)
Comment (NeoxAzrot) [reply components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx:17]: @alexbeno

File: app/migration/projects/context/project.context.type.ts (hunk 35 above)
[the remark this answers is not in the document]
Comment (alexbeno) [reply app/migration/projects/context/project.context.type.ts]: agree we need to convert into union type

File: components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx (hunk 22 above)
Comment (alexbeno) [reply components/features/project-lead-Invitation-banner/project-lead-Invitation-banner.tsx]: rename on into "location"

**Pull request merged**